src/App.js

```diff
@@ -1,25 +1,36 @@
-import logo from './logo.svg';
+import React from 'react';
 import './App.css';
+import Counter from './components/Counter'
+import { Button } from '@material-ui/core';
+class App extends React.Component {
```

Comment:
I think this is fine for now, but I'd really really really recommend we stick to only using functional components instead of class based components bc they're wayyyy easier to debug with less boilerplate and most react dev's today view class based components as a legacy system (even if FB hasn't deprecated it)

Reply:
I see. So for components without internal state, we should use [functional components](https://reactjs.org/docs/components-and-props.html)
But for components that do require state, we continue to use [class components](https://reactjs.org/docs/state-and-lifecycle.html), correct?

Reply:
Close. Functional components can do every thing a class component can do, but better (using hooks).

So what I'm saying is to always use functional components and never use class components. 

For example, I was able to rewrite `App.js` from a class component to a functional component and it is much more modern, understandable, and maintainable. 
This example turned 37 lines of code to 23 which isn't a big deal, but you can see how this would be a much bigger deal with a more involved component.

```javascript
import React, { useState } from 'react';
import './App.css';
import Counter from './components/Counter'
import { Button } from '@material-ui/core';

const App = () => {
  /*
   * count is our state variable
   * count will have a default state of 0
   * couunt will be updated using the setCount function
   */
  const [count, setCount] = useState(0);

  const incrementer = () => setCount(count + 1);

  return (
    <>
      <div>
        Hello {this.props.name}
      </div>
      <Counter count={ this.state.count }/>
      <Button onClick={incrementer} >Primary</Button>
    </>
  );
}

export default App;
```

Reply:
Edited the code above because there was an error since I was still using the `this` keyword. 
The above should now work properly lol

Reply:
I think we should adopt this into this PR. I'll make the changes later to use functional components. Def agree it's much cleaner/clearer than class components. Thanks!